WordPress/src/main/java/org/wordpress/android/ui/mysite/SelectedSiteSource.kt

```diff
@@ -17,26 +18,24 @@ import javax.inject.Singleton
 class SelectedSiteSource @Inject constructor(
     private val selectedSiteRepository: SelectedSiteRepository,
     private val dispatcher: Dispatcher
-) : MySiteSource<SelectedSite> {
-    val refresh: MutableLiveData<Boolean> = MutableLiveData<Boolean>(false)
+) : MySiteRefreshSource<SelectedSite> {
+    override val refresh: MutableLiveData<Boolean> = MutableLiveData<Boolean>(false)
 
-    override fun buildSource(
+    override fun build(
         coroutineScope: CoroutineScope,
         siteLocalId: Int
     ) = selectedSiteRepository.selectedSiteChange
             .filter { it == null || it.id == siteLocalId }
             .map { SelectedSite(it) }
 
-    fun refresh() {
+    override fun refresh() {
         selectedSiteRepository.updateSiteSettingsIfNecessary()
         selectedSiteRepository.getSelectedSite()?.let {
             refresh.postValue(true)
             dispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(it))
         }
     }
 
-    fun isRefreshing() = refresh.value as Boolean
-
     init {
         dispatcher.register(this)
     }
```

Comment:
Suggestion (💡): During `onSiteChanged(...)` the `refresh.value = false` is used directly. How about adding an new `onRefreshed()` interface implementation which will be used in such cases instead?

```
        fun onRefreshed() {
            refresh.value = false
        }
```

As with my above comment, what I am trying to avoid updating the mutable `refresh` live data on the implementing clients. Wdyt?

Reply:
I like this one!! Adding it